Build errors with node `v16.15.1` (npm `v8.11.0`)

Build fails with the latest node `v16.15.1` and npm `v8.11.0`. 
I am afraid the only way to mitigate this seems to be downgrading to node `v16.14.2` with npm `v8.5.5`.

By chance, there was [a commit that fixes this in `react-bootstrap`](https://github.com/react-bootstrap/react-bootstrap/pull/6383) about ~20 hours ago. Should be resolved with the next version of `react-bootstrap`.

Well here is the issue: https://github.com/rootzoll/raspiblitz/issues/3224

TG report: https://t.me/JoinMarketWebUI/3383
```
npm ERR! notsup Required: {"node":">=16.11.0","npm":">=8.0.0"}
```
Updated to the latest NodeJS v18.5 and now running into this with the latest Raspiblitz scripts on an RPi:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: moving-letters@1.0.1
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   dev react@"^17.0.2" from the root project
npm ERR!   peer react@"^16.8.0 || ^17.0.0-rc.1" from @react-aria/ssr@3.1.2
npm ERR!   node_modules/@react-aria/ssr
npm ERR!     @react-aria/ssr@"^3.0.1" from @restart/ui@1.2.0
npm ERR!     node_modules/@restart/ui
npm ERR!       @restart/ui@"^1.2.0" from react-bootstrap@2.4.0
npm ERR!       node_modules/react-bootstrap
npm ERR!         dev react-bootstrap@"^2.4.0" from the root project
npm ERR!   14 more (@restart/hooks, @restart/ui, @testing-library/react, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.0.0 || ^16.0.0" from moving-letters@1.0.1
npm ERR! node_modules/moving-letters
npm ERR!   dev moving-letters@"^1.0.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@16.14.0
npm ERR! node_modules/react
npm ERR!   peer react@"^15.0.0 || ^16.0.0" from moving-letters@1.0.1
npm ERR!   node_modules/moving-letters
npm ERR!     dev moving-letters@"^1.0.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/joinmarket/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/joinmarket/.npm/_logs/2022-07-15T18_48_31_566Z-debug-0.log
FAIL - npm install did not run correctly, aborting
```

Is this expected to be fixed or should we ship with some other node version?

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: ['16.11.0']
+        node-version: ['16.16.0']
 
     steps:
       - name: Checkout

--- a/package.json
+++ b/package.json
@@ -34,7 +34,7 @@
     "prettier": "^2.6.2",
     "qrcode": "^1.5.0",
     "react": "^17.0.2",
-    "react-bootstrap": "^2.4.0",
+    "react-bootstrap": "^2.5.0-beta.1",
     "react-dom": "^17.0.2",
     "react-i18next": "^11.16.9",
     "react-router-bootstrap": "^0.26.1",

--- a/package.json
+++ b/package.json
@@ -37,7 +37,7 @@
     "react-bootstrap": "^2.5.0-beta.1",
     "react-dom": "^17.0.2",
     "react-i18next": "^11.16.9",
-    "react-router-bootstrap": "^0.26.1",
+    "react-router-bootstrap": "^0.26.2",
     "react-router-dom": "^6.3.0",
     "react-scripts": "^5.0.1",
     "typescript": "^4.7.2",

--- a/package.json
+++ b/package.json
@@ -31,7 +31,6 @@
     "jest-watch-typeahead": "^0.6.5",
     "jest-websocket-mock": "^2.3.0",
     "lint-staged": "^12.4.2",
-    "moving-letters": "^1.0.1",
     "prettier": "^2.6.2",
     "qrcode": "^1.5.0",
     "react": "^17.0.2",

--- a/package.json
+++ b/package.json
@@ -19,7 +19,7 @@
     "@types/node": "^17.0.35",
     "@types/react": "^17.0.43",
     "@types/react-dom": "^17.0.14",
-    "bootstrap": "^5.1.3",
+    "bootstrap": "^5.2.0",
     "buffer": "^6.0.3",
     "classnames": "^2.3.1",
     "conventional-changelog": "^3.1.25",

--- a/src/components/Earn.jsx
+++ b/src/components/Earn.jsx
@@ -91,7 +91,7 @@ export default function Earn() {
   const serviceInfo = useServiceInfo()
   const reloadServiceInfo = useReloadServiceInfo()
 
-  const [showAdvancedSettings, setShowAdvancedSettings] = useState(false)
+  const [showSettings, setShowSettings] = useState(false)
   const [alert, setAlert] = useState(null)
   const [serviceInfoAlert, setServiceInfoAlert] = useState(null)
   const [isLoading, setIsLoading] = useState(true)
@@ -333,17 +333,17 @@ export default function Earn() {
                       <rb.Button
                         variant={`${settings.theme}`}
                         className={`${styles['settings-btn']} d-flex align-items-center`}
-                        onClick={() => setShowAdvancedSettings((current) => !current)}
+                        onClick={() => setShowSettings((current) => !current)}
                       >
                         {t('earn.button_settings')}
                         <Sprite
-                          symbol={`caret-${showAdvancedSettings ? 'up' : 'down'}`}
+                          symbol={`caret-${showSettings ? 'up' : 'down'}`}
                           className="ms-1"
                           width="20"
                           height="20"
                         />
                       </rb.Button>
-                      {showAdvancedSettings && (
+                      {showSettings && (
                         <div className="my-4">
                           <rb.Form.Group className="mb-4 d-flex justify-content-center" controlId="offertype">
                             <SegmentedTabs
@@ -482,33 +482,35 @@ export default function Earn() {
                       <hr className="m-0" />
                     </div>
                   )}
-                  <rb.Button
-                    variant="dark"
-                    type="submit"
-                    className="mt-4"
-                    disabled={isLoading || isSubmitting || isWaitingMakerStart || isWaitingMakerStop}
-                  >
-                    <div className="d-flex justify-content-center align-items-center">
-                      {(isWaitingMakerStart || isWaitingMakerStop) && (
-                        <rb.Spinner
-                          as="span"
-                          animation="border"
-                          size="sm"
-                          role="status"
-                          aria-hidden="true"
-                          className="me-2"
-                        />
-                      )}
-                      {isWaitingMakerStart || isWaitingMakerStop ? (
-                        <>
-                          {isWaitingMakerStart && t('earn.text_starting')}
-                          {isWaitingMakerStop && t('earn.text_stopping')}
-                        </>
-                      ) : (
-                        <>{serviceInfo?.makerRunning === true ? t('earn.button_stop') : t('earn.button_start')}</>
-                      )}
-                    </div>
-                  </rb.Button>
+                  <div class="mt-4">
+                    <rb.Button
+                      variant="dark"
+                      type="submit"
+                      className={styles['earn-btn']}
+                      disabled={isLoading || isSubmitting || isWaitingMakerStart || isWaitingMakerStop}
+                    >
+                      <div className="d-flex justify-content-center align-items-center">
+                        {(isWaitingMakerStart || isWaitingMakerStop) && (
+                          <rb.Spinner
+                            as="span"
+                            animation="border"
+                            size="sm"
+                            role="status"
+                            aria-hidden="true"
+                            className="me-2"
+                          />
+                        )}
+                        {isWaitingMakerStart || isWaitingMakerStop ? (
+                          <>
+                            {isWaitingMakerStart && t('earn.text_starting')}
+                            {isWaitingMakerStop && t('earn.text_stopping')}
+                          </>
+                        ) : (
+                          <>{serviceInfo?.makerRunning === true ? t('earn.button_stop') : t('earn.button_start')}</>
+                        )}
+                      </div>
+                    </rb.Button>
+                  </div>
                 </rb.Form>
               )}
             </Formik>

--- a/src/components/Earn.module.css
+++ b/src/components/Earn.module.css
@@ -37,11 +37,11 @@
 }
 
 .earn hr {
-  background-color: rgba(222, 222, 222, 1);
+  border-color: rgba(222, 222, 222, 1);
   opacity: 100%;
 }
 
 :root[data-theme='dark'] .earn hr {
-  background-color: var(--bs-gray-800);
+  border-color: var(--bs-gray-800);
   opacity: 100%;
 }

--- a/src/components/Earn.module.css
+++ b/src/components/Earn.module.css
@@ -12,7 +12,7 @@
   height: 3.5rem;
 }
 
-.earn form button {
+.earn .earn-btn {
   height: 3rem;
   width: 100%;
 }
@@ -25,7 +25,8 @@
   background-color: transparent !important;
   border: none;
   padding-left: 0;
-  font-weight: 500;
+  height: 3rem;
+  width: 100%;
 }
 
 .settings-container .input-group-text {

--- a/src/components/Receive.jsx
+++ b/src/components/Receive.jsx
@@ -112,7 +112,7 @@ export default function Receive() {
           <rb.Button
             variant={`${settings.theme}`}
             className={`${styles['settings-btn']} d-flex align-items-center`}
-            onClick={() => setShowSettings(!showSettings)}
+            onClick={() => setShowSettings((current) => !current)}
           >
             {t('receive.button_settings')}
             <Sprite symbol={`caret-${showSettings ? 'up' : 'down'}`} className="ms-1" width="20" height="20" />
@@ -163,10 +163,10 @@ export default function Receive() {
               </rb.Form.Group>
             </div>
           )}
-          <hr />
+          <hr className="m-0" />
         </div>
 
-        <div className="mt-2 d-flex justify-content-center">
+        <div className="mt-4 d-flex justify-content-center">
           <rb.Button
             variant="outline-dark"
             type="submit"

--- a/src/components/Receive.module.css
+++ b/src/components/Receive.module.css
@@ -22,6 +22,8 @@
   background-color: transparent !important;
   border: none;
   padding-left: 0;
+  height: 3rem;
+  width: 100%;
 }
 
 .receive form input {

--- a/src/components/Receive.module.css
+++ b/src/components/Receive.module.css
@@ -1,10 +1,10 @@
 .receive hr {
-  background-color: rgba(222, 222, 222, 1);
+  border-color: rgba(222, 222, 222, 1);
   opacity: 100%;
 }
 
 :root[data-theme='dark'] .receive hr {
-  background-color: var(--bs-gray-800);
+  border-color: var(--bs-gray-800);
   opacity: 100%;
 }
 

--- a/src/components/Send.module.css
+++ b/src/components/Send.module.css
@@ -95,7 +95,7 @@ input[type='number'] {
   color: var(--bs-white) !important;
 }
 
-:root[data-theme='dark'] .sweep-breakdoown-paragraph {
+:root[data-theme='dark'] .sweep-breakdown-paragraph {
   color: var(--bs-white) !important;
 }
 

--- a/src/index.css
+++ b/src/index.css
@@ -609,7 +609,7 @@ h2 {
 }
 
 :root[data-theme='dark'] .link-dark {
-  color: var(--bs-white);
+  color: var(--bs-white) !important;
 }
 
 :root[data-theme='dark'] .toast {

